zeppelin-server/src/main/java/org/apache/zeppelin/socket/NotebookServer.java

```diff
@@ -1515,7 +1515,7 @@ private InterpreterGroup findInterpreterGroupForParagraph(Note note, String para
     if (paragraph == null) {
       throw new IllegalArgumentException("Unknown paragraph with id : " + paragraphId);
     }
-    return paragraph.getCurrentRepl().getInterpreterGroup();
+    return paragraph.getInterpreter().getInterpreterGroup();
```

Comment:
I think `getInterpreter` is a bit generic - as we already have `getInterpreter` in other classes and their purposes are very different from this?

Reply:
Change it to `getBindedInterpreter`